### PR TITLE
document and clean the AST code

### PR DESCRIPTION
**Pull request by performantdata**
Documentation and clean-up of IDE warnings related to my investigation into separating the AST package into its own module.

### CODE REVIEW

File: modules/core/src/main/scala/sangria/ast/QueryAst.scala [hunk 1]
```diff
@@ -498,6 +514,8 @@ case class Comment(text: String, location: Option[AstLocation] = None) extends A
 
 // Schema Definition
 
+/** @group typesystem
```
Comment (yanns) [inline modules/core/src/main/scala/sangria/ast/QueryAst.scala:517]: What is this `@group`? is if the scaladoc?

**Review by yanns — APPROVED** [review]
💯

File: modules/core/src/main/scala/sangria/ast/QueryAst.scala (hunk 1 above)
Comment (performantdata) [reply modules/core/src/main/scala/sangria/ast/QueryAst.scala:517]: https://docs.scala-lang.org/overviews/scaladoc/for-library-authors.html#member-grouping-tags

Comment (performantdata) [reply modules/core/src/main/scala/sangria/ast/QueryAst.scala:517]: You can already see an example [here](https://www.javadoc.io/doc/org.sangria-graphql/sangria_2.13/latest/sangria/ast/index.html).

**Pull request merged**